src/main/java/edu/cornell/kfs/krad/util/MaintenanceLockUtils.java

```diff
@@ -0,0 +1,52 @@
+package edu.cornell.kfs.krad.util;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.sys.KFSConstants;
+
+public final class MaintenanceLockUtils {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final int BULK_LOCK_QUERY_LIMIT = 500;
+
+    private MaintenanceLockUtils() {
+        throw new UnsupportedOperationException("do not call");
+    }
+
+    public static String doChunkedSearchForAnyLockingDocumentNumber(List<String> lockingRepresentations,
+            String documentNumber, BiFunction<List<String>, String, String> actualQueryOperation) {
+        if (CollectionUtils.isEmpty(lockingRepresentations)) {
+            LOG.debug("doChunkedSearchForAnyLockingDocumentNumber, No locking reps specified, skipping search");
+            return null;
+        }
+        LOG.debug("doChunkedSearchForAnyLockingDocumentNumber, Searching with {} lock reps{}",
```

Comment:
should this block be in an IF statement LOG.isiDebugEnabled() so the string is only built if the log statement needs to be generated

Reply:
interesting convention, but we should follow suit as you have done